pkg/controller/operands/operand.go

```diff
@@ -93,7 +95,43 @@ func (h *genericOperand) ensure(req *common.HcoRequest) *EnsureResult {
 	found := h.hooks.getEmptyCr()
 	err = h.Client.Get(req.Ctx, key, found)
 	if err != nil {
-		return h.createNewCr(req, err, cr, res)
+		result := h.createNewCr(req, err, cr, res)
+
+		if apierrors.IsAlreadyExists(result.Err) {
+			// we failed trying to create it due to a caching error
+			// or we neither tried because we know that the object is already there for sure,
+			// but we cannot get it due to a bad cache hit.
+			// Let's try updating it bypassing the client cache mechanism
+			cfg, configerr := config.GetConfig()
+			if configerr != nil {
+				req.Logger.Error(configerr, "failed creating a config for a custom client")
+				return &EnsureResult{
+					Err: configerr,
+				}
+			}
+			apiClient, acerr := client.New(cfg, client.Options{
+				Scheme: h.Scheme,
+			})
+			if acerr != nil {
+				req.Logger.Error(acerr, "failed creating a custom client to bypass the cache")
+				return &EnsureResult{
+					Err: acerr,
+				}
+			}
+			geterr := apiClient.Get(req.Ctx, key, found)
+			if geterr != nil {
+				req.Logger.Error(geterr, "failed trying to get the object bypassing the cache")
+				return &EnsureResult{
+					Err: geterr,
+				}
+			}
+			originalClient := h.Client
+			h.Client = apiClient
+			existingcrresult := h.handleExistingCr(req, key, found, cr, res)
+			h.Client = originalClient
+			return existingcrresult
+		}
```

Comment:
1. Can you put all of this in a new function? the original is too long now.
2. I wonder (not urgent) if we can change it to be "thread safe" if we'll want to call the operand ensure methods in their own goroutine. WDYT?

Reply:
done